examples/plot_dl_diabetes_confidence.py

```diff
@@ -0,0 +1,181 @@
+"""
+Coefficient estimates with Desparsified Lasso on the diabetes dataset
+=====================================================================
+
+This example illustrates how to compute de-biased coefficient estimates and confidence
+intervals using :class:`~hidimstat.DesparsifiedLasso` on the diabetes dataset.
+This example is inspired by :footcite:t:`hastie2015statistical`.
+
+While the L1 penalty used in Lasso regression is a powerful regularization technique for
+building predictive models, it introduces a bias in the coefficient estimates (shrinkage).
+When the goal is to interpret the importance of features or perform inference, this bias
+should be corrected. The Desparsified Lasso provides a method to obtain unbiased coefficient
+estimates, along with confidence intervals and p-values for hypothesis testing.
+Read more in the :ref:`User Guide <slm_methods>`.
+"""
+
+# %%
+# Load diabetes dataset
+# ---------------------
+# The diabetes dataset is a well-known benchmark for regression tasks. It
+# contains 10 features corresponding to baseline measurements and a quantitative
+# measure of disease progression.
+
+from sklearn.datasets import load_diabetes
+
+data = load_diabetes(as_frame=True)
+X = data["data"].to_numpy()
+y = data["target"].to_numpy()
+feature_names = data["data"].columns.tolist()
+data["data"].head()
+
+
+# %%
+# Add spurious features
+# ---------------------
+# To evaluate the feature selection capabilities of the Desparsified Lasso, we
+# artificially add spurious features. These are constructed as random linear
+# combinations of the original features plus noise, ensuring they are correlated
+# with the predictors but have no true association with the target variable.
+
+import numpy as np
+from sklearn.preprocessing import StandardScaler
+
+X = StandardScaler().fit_transform(X)
+y = StandardScaler().fit_transform(y.reshape(-1, 1)).ravel()
+
+seed = 0
+rng = np.random.default_rng(seed=seed)
+n_spurious = 5
+X_spurious_list = []
+for i in range(n_spurious):
+    X_spurious = (
+        X[:, rng.choice(X.shape[1], size=3, replace=False)]
+        + 1 * rng.normal(size=X[:, :3].shape)
+    ).sum(axis=1, keepdims=True)
+    X_spurious_normalized = StandardScaler().fit_transform(X_spurious)
+    X_spurious_list.append(X_spurious_normalized)
+    feature_names.append(f"spurious_{i}")
+X = np.hstack([X] + X_spurious_list)
+
+
+# %%
+# Feature importance with Desparsified Lasso
+# ------------------------------------------
+# We fit the Desparsified Lasso on the dataset to obtain de-biased coefficient
+# estimates and 95% confidence intervals.
+
```

Comment:
I think that before performing prediction, we should evaluate the R^2 in predicting y + the feature correlation matrix

Reply:
This should actually be the case for all examples !